SIRF_data_preparation/evaluation_utilities.py

```diff
@@ -15,16 +15,31 @@ def read_objectives(datadir='.'):
     with (Path(datadir) / 'objectives.csv').open() as csvfile:
         reader = csv.reader(csvfile)
         next(reader) # skip first (header) line
-        return numpy.asarray([tuple(map(float, row)) for row in reader])
+        return np.asarray([tuple(map(float, row)) for row in reader])
 
 
 def get_metrics(qm: QualityMetrics, iters: Iterator[int], srcdir='.'):
     """Read 'iter_{iter_glob}.hv' images from datadir, compute metrics and return as 2d array"""
-    return numpy.asarray([
+    return np.asarray([
         list(qm.evaluate(STIR.ImageData(str(Path(srcdir) / f'iter_{i:04d}.hv'))).values()) for i in iters])
 
 
-def plot_metrics(iters: Iterator[int], m: numpy.ndarray, labels=None, suffix=""):
+def pass_index(metrics: np.ndarray, thresh: np.ndarray, window: int = 1) -> int:
+    """
+    Returns first index of `metrics` with value <= `thresh`.
+    The value must remain below the threshold for at least `window`.
+    Raises IndexError if doesn't pass.
+    """
+    assert metrics.shape[1] == len(thresh)
+    assert metrics.ndim == 2
+    assert thresh.ndim == 1
+
+    m = (metrics <= thresh[None]).all(1)
+    assert window == 1
```

Comment:
I guess this is a future feature? Fine for me

Reply:
Added implementation.